### DISCUSSION TITLE
Use region start/end as good/bad starting commits when starting git bisect from magit-log

### DISCUSSION THREAD

**Discussion by ashton314** [magit/magit#5049] [Ideas]
Suppose I'm looking at my Git log. I can select a bunch of commits and use something like `d r` to get a diff between the top and bottom of the selection.

I think the same for git bisect would be cool: select a region, the earliest commit is assumed to be the default "good" commit, and the newest commit is assumed to be the "bad" commit.

Right now when I do `B B` (bisect start) the current point is the first option shown when asked for the bad commit in my `completing-read` interface. (Vertico fwiw) I'd like the mark to be the first option for the good commit in the `completing-read` interface as well. (Right now it defaults to HEAD I think.)

**Comment by kyleam** [comment]
That proposed behavior seems reasonable to me, perhaps given some checks (see snippet).

I don't think I'd find it useful myself because, while I bisect fairly often, it's not typically over ranges of commits that would fit comfortably on a screen to select, and inspecting the log typically isn't my starting point. (And of course part of the appeal of a bisect is that it doesn't cost much to be loose with the boundaries.)

Fwiw you could get this behavior on your end with something like

```elisp
(advice-add
 'magit-bisect-start-read-args :around
 (lambda (f)
   (if-let* ((commits (magit-region-values '(commit branch) t)))
       (let ((b (car commits))
             (a (last commits))
             (args (transient-args 'magit-bisect)))
         (cond ((magit-rev-ancestor-p a b)
                (list b a args))
               ((magit-rev-ancestor-p b a)
                (list a b args))
               (t (funcall f))))
     (funcall f)))
 '((name . "magit-bisect-region-args")))
```

**Comment by tarsius** [comment]
I've added this in fe56c0d10a91d00aed65f831b8435a7224bc7730.
